cmake/Modules/ObsDefaults_Windows.cmake

```diff
@@ -0,0 +1,80 @@
+cmake_minimum_required(VERSION 3.16)
```

Comment:
I don’t but haven’t gotten input about making a more recent version the default on Windows.

Reply:
We've bumped the CMake minimum version across all systems from 2.8.12 to 3.10 in https://github.com/obsproject/obs-studio/pull/2081, partly because "CMake 3.10 is required for CXX_STANDARD VS support.", which I note we've removed in this PR.  We then bumped the CMake minimum version _only for Windows_ in https://github.com/obsproject/obs-studio/pull/2208 for WinRT.  From @jpark37 :
> yeah, 3.10 is only really necessary for windows. that's when setting C++17 standard affected VC++ projects.
> jim already added a windows only required version check for 3.16 somewhere to deal with the PCH in the new winrt DLL.

If we can set them independently, I'm personally fine with requiring the latest stable release on Windows and macOS where that makes sense, unless that somehow causes issues or a massive disparity for Linux.  See also, our other conversation on using the CMake default version for Ubuntu versus using CMake from Kitware's apt repo.

Reply:
To clarify my viewpoint:  I don't think it's a burdensome requirement on Windows and macOS to require CMake 3.20 as the minimum required version.  Though, I suppose it would be easier to justify the bump from 3.16 if there was some feature of CMake in newer versions that we wanted/needed.